src/geneweaver/api/services/publications.py

```diff
@@ -22,17 +26,52 @@ def get_publication(cursor: Cursor, pub_id: int) -> dict:
     return pub
 
 
-def get_publication_by_pubmed_id(cursor: Cursor, pub_med_id: str) -> dict:
+def get_publication_by_pubmed_id(cursor: Cursor, pubmed_id: str) -> dict:
     """Get a publication by Pubmed Id from the DB.
 
     @param cursor: DB cursor
-    @param pub_med_id: pub med identifier
+    @param pubmed_id: pub med identifier
     @return: dictionary response (publication).
     """
     try:
-        pub = db_publication.by_pubmed_id(cursor, pub_med_id)
+        pub = db_publication.by_pubmed_id(cursor, pubmed_id)
         return pub
 
     except Exception as err:
         logger.error(err)
         raise err
+
+
+def add_pubmed_record(cursor: Cursor, user: User, pubmed_id: str) -> dict:
+    """Add pubmed publication to the DB given pub-med id.
+
+    @param cursor: DB cursor
+    @param user: logged-in user
+    @param pubmed_id: pubmed id
+    @return:
+    """
+    try:
+        # check logged-in user
+        if user is None or user.id is None:
+            return {"error": True, "message": message.ACCESS_FORBIDDEN}
+        # check publication is not already in the DB
+        pub = db_publication.by_pubmed_id(cursor, pubmed_id)
+        if pub:
+            return {"error": True, "message": message.RECORD_EXISTS}
```

Comment:
If the publication exists, you can just return it. That would make the endpoint idempotent, which is desirable.